### PR TITLE
[WIP] move 'x11' to 'drivers/' directory

### PR DESCRIPTION
**Pull request by hpvb**
This PR implements a new 'freedesktop' platform and moves the X11 specific code to 'drivers/'

### CODE REVIEW

**Comment by reduz** [comment]
sorry, I know you will kill me for this,
but In the end, I kind of like naming the platform "freedesktop"

**Comment by hpvb** [comment]
I know this will easily devolve into a bikeshedding exercise but I'd like to offer one argument against the 'freedesktop' name, even though I originally came up with it: Splitting out the X11, Wayland, and Dummy renderers and audio drivers and whatnot should eventually lead to the killing of the 'server' platform entirely. At that point building a server binary with a 'freedesktop' target seems to make little sense.

If you don't think this is important I'll rename the platform and this is the last that has been said of that :)

**Comment by akien-mga** [comment]
I think "freedesktop" doesn't make sense for most people who are not very familiar with the Linux and *BSD ecosystems. I don't see Windows users with custom libraries bothering supporting a "freedesktop" platform with custom builds, while "linuxbsd" is already clearer about what is being supported in end user terms.

File: COPYRIGHT.txt [hunk 1]
```diff
@@ -88,7 +88,7 @@ License: Apache-2.0
 Files: ./platform/android/power_android.cpp
  ./platform/osx/power_osx.cpp
  ./platform/windows/power_windows.cpp
- ./platform/x11/power_x11.cpp
+ ./platform/linuxbsd/power_x11.cpp
```
Comment (akien-mga) [inline COPYRIGHT.txt]: You should move it up to keep the alphabetical ordering.

**Comment by hpvb** [comment]
@akien-mga well, Wayland is a freedesktop.org project. :)

I like.... pink 🤣

**Comment by akien-mga** [comment]
BTW, eventually (not in this PR) we'll have to add support for multiple export platforms sharing the same platform code.

Currently to export to FreeBSD, you have to override the Linux X11 official template with your custom FreeBSD one - ideally we'd give FreeBSD and OpenBSD the possibility to have their own "FreeBSD (X11)"/"FreeBSD (Wayland)"/etc. entries in the export manager. Whether we should ship official templates for them is to be determined, but we should at least make it possible to export to those platforms without hacks.

**Comment by hpvb** [comment]
@akien-mga I agree, I'll see if I can do some work on that also. Please note however that there will not be separate export templates for X11 and Wayland. This will be determined at runtime (like it works for SDL, GTK, and QT). We will not link to Wayland or X11 libraries directly anymore on the released binaries but instead dlopen() the libraries when they are required. Wayland/X11 libraries will be build-time required but runtime optional.

**Comment by akien-mga** [comment]
> Please note however that there will not be separate export templates for X11 and Wayland. This will be determined at runtime (like it works for SDL, GTK, and QT).

That's perfect :)

**Comment by toger5** [comment]
Any status / updates on this?

**Comment by akien-mga** [comment]
Closing as this will have to be redone from scratch, probably for 3.2 as we're running out of time for 3.1.
It's still a wanted change though and @hpvb will be working on it when it's back in our priorities.

**Comment by toger5** [comment]
@akien-mga you mentioned it is necessary to implement this from scratch. For an experiment @wmww and I really want that to work. Currently merging that code on master and trying to make it work. Was there something entirely wrong with the implementation from this PR? Or does it just need some love to work with current master?

**Pull request closed without merging**